cmd/pulumi-test-language/providers/simple_invoke_provider.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package providers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/blang/semver"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+type SimpleInvokeProvider struct {
+	plugin.UnimplementedProvider
+}
+
+var _ plugin.Provider = (*SimpleInvokeProvider)(nil)
+
+func (p *SimpleInvokeProvider) Close() error {
+	return nil
+}
+
+func (p *SimpleInvokeProvider) Configure(
+	context.Context, plugin.ConfigureRequest,
+) (plugin.ConfigureResponse, error) {
+	return plugin.ConfigureResponse{}, nil
+}
+
+func (p *SimpleInvokeProvider) Pkg() tokens.Package {
+	return "simple-invoke"
+}
+
+func (p *SimpleInvokeProvider) GetPluginInfo(context.Context) (workspace.PluginInfo, error) {
+	ver := semver.MustParse("10.0.0")
+	return workspace.PluginInfo{
+		Version: &ver,
+	}, nil
+}
+
+func (p *SimpleInvokeProvider) GetSchema(
+	context.Context, plugin.GetSchemaRequest,
+) (plugin.GetSchemaResponse, error) {
+	pkg := schema.PackageSpec{
+		Name:    "simple-invoke",
+		Version: "10.0.0",
+		Resources: map[string]schema.ResourceSpec{
+			// A small resource that just has a single string property.
+			"simple-invoke:index:StringResource": {
+				ObjectTypeSpec: schema.ObjectTypeSpec{
+					Type: "object",
+					Properties: map[string]schema.PropertySpec{
+						"text": {
+							TypeSpec: schema.TypeSpec{
+								Type: "string",
+							},
+						},
+					},
+					Required: []string{"text"},
+				},
+			},
+		},
+		Functions: map[string]schema.FunctionSpec{
+			"simple-invoke:index:myInvoke": {
```

Comment:
Interestingly if you name this "MyInvoke" (i.e. pascal case like we do resources) it breaks python codegen.